docs/test-fixtures.md

```diff
@@ -30,8 +47,59 @@ class UniqueTestsFixture {
  }
 ```
 
-The two test cases here will create uniquely-named derived classes of UniqueTestsFixture and thus can access the `getID()` protected method and `conn` member variables. This ensures that both the test cases are able to create a DBConnection using the same method (DRY principle) and that any ID's created are unique such that the order that tests are executed does not matter.
+The two test cases here will create uniquely-named derived classes of UniqueTestsFixture and thus can access the `getID()` protected method and `conn` member variables. This ensures that both the test cases are able to create a DBConnection using the same method (DRY principle) and that any ID's created are unique such that the order that tests are executed does not matter. 
+
+### 2. `METHOD_AS_TEST_CASE`
+
+`METHOD_AS_TEST_CASE` lets you register a member function of a class as a Catch2 test case. The class will be separately instantiated for each method registered in this way.
+
+```cpp
+class TestClass {
+    std::string s;
+
+public:
+    TestClass()
+        :s( "hello" )
+    {}
+
+    void testCase() {
+        REQUIRE( s == "hello" );
+    }
+};
+
+
+METHOD_AS_TEST_CASE( TestClass::testCase, "Use class's method as a test case", "[class]" )
+```
+
+This type of fixture is similar to [TEST_CASE_METHOD](#1-test_case_method) except in this case it will directly use the provided class to create an object rather than a derived class.
+
+### 3. `TEST_CASE_FIXTURE`
+
+> [Introduced](link-to-issue-or-PR) in Catch2 X.Y.Z
+
+`TEST_CASE_FIXTURE` behaves in the same way as [TEST_CASE_METHOD](#1-test_case_method) except that there will only be one instance created throughout the entire run of a test case. To demonstrate this have a look at the following example:
+
+```cpp
+struct MyFixture{
+    int MyInt = 0;
+};
+
+TEST_CASE_FIXTURE(MyFixture, "Tests with MyFixture") { 
+    
+    const int val = MyInt++;
+
+    SECTION("First partial run") {
+        REQUIRE(val == 0);
+    }
+
+    SECTION("Second partial run") {
+        REQUIRE(val == 1);
+    }
+}
+```
+This test case will be executed twice as there are two leaf sections. On the first run `val` will be `0` and on the second run `val` will be `1`. This is useful if you would like to share some expensive setup code with all runs of your test case which can't be done at static initialization time.
```

Comment:
I am not particularly happy about the example/explanation combo. My first reaction was "well, how's that different from using static variable here".

I can't give you better version right now, but I think this should be focused on the performance difference from `TEST_CASE_METHOD`, maybe have something like

```cpp
// Pretend doing hard work
std::this_thread::sleep_for(2s);
```

in the fixture constructor and explain that `TEST_CASE_FIXTURE` takes 2 seconds, while `TEST_CASE_METHOD` needs 2/4/6/8/... seconds to execute.

Reply:
This should probably also explain that two different `TEST_CASE_FIXTURE`s have **separate** instances of fixture, i.e., the fixture is not static globally, just for the paths defined in one `TEST_CASE`.

Reply:
I have redone the example and the documentation for this feature now. I feel like it now better highlights why a test writer would use this over TEST_CASE_METHOD.